scripts/upgrade.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+__ScriptName="$(basename "$0")"
+SALT_CALL=${SALT_CALL:-salt-call}
+
+VERBOSE=${VERBOSE:-0}
+LOGFILE=/var/log/metalk8s-upgrade.log
+TO_VERSION=0
+UPGRADE=0
+PRECHECK=0
+
+# Some helpful guide on upgrade options
+__usage() {
+    cat << EOF
+  Usage :  ${__ScriptName} -t <destination_version> [options]
+  Options:
+    -h | --help              Display HELP options
+    -p | --pre-check         Show Pre-Check output.
+    -t | --to                Upgrade to specified version
+    -v | --verbose           Run in verbose mode
+EOF
+}
+
+# # handle the default case if script is called with no args
+[ "$#" -eq 0 ] && __usage
+
+if ! options=$(getopt --options p:vht: --long pre-check:,verbose,help,to: -- "$@"); then
+    echo 1>&2 "Incorrect arguments: Provide a valid MetalK8s version"
+    exit 1
+fi
+eval set -- "$options"
+
+while true; do
+    case "$1" in
+        -p|--pre-check)
+            VERBOSE=1;
+            TO_VERSION="$2"
+            PRECHECK=1
+            shift 2 ;;
+        -v|--verbose)
+            VERBOSE=1;
+            shift;;
+        -t|--to)
+            TO_VERSION="$2"
+            UPGRADE=1
```

Comment:
This one should only store the version that it. And after you will execute something in function if you have `DRY_RUN` set to `1` or not